packages/probing-frontend/src/pages/MonitoringDetailPage/MonitoringDetail.page.tsx

```diff
@@ -0,0 +1,76 @@
+import { MonitoringQueries } from '@/api/monitoring/monitoring.hooks'
+import { Link, useParams } from '@/router'
+import { MonitoringEserviceTelemetry } from './components/MonitoringEserviceTelemetry'
+import { MonitoringEserviceProbing } from './components/MonitoringEserviceProbing'
+import { MonitoringEserviceDetail } from './components/MonitoringEserviceDetail'
+import { Box } from '@mui/system'
+import { useTranslation } from 'react-i18next'
+import ArrowBackIcon from '@mui/icons-material/ArrowBack'
+import { Typography } from '@mui/material'
+import { useLoadingOverlay } from '@/stores'
+
+export const MonitoringDetailPage = () => {
+  const { t } = useTranslation('common', {
+    keyPrefix: 'detailsPage',
+  })
+
+  const { showOverlay, hideOverlay } = useLoadingOverlay()
+  const { id: eserviceId } = useParams<'MONITORING_DETAIL'>()
+  const { data: eservicesDetail } = MonitoringQueries.useGetEserviceData(
+    {
+      eserviceId,
+    },
+    { suspense: false }
+  )
+
+  const {
+    data: eservicesProbingDetail,
+    isSuccess: isSuccessProbing,
+    refetch,
+  } = MonitoringQueries.useGetEserviceProbingData({ eserviceId }, { suspense: false })
+
+  const hasValidData = isSuccessProbing && eservicesDetail
+
+  const handleRefetch = async () => {
+    showOverlay(t('loading'))
+    // We want show the loading overlay for at least 1 second, to avoid flickering
+    await Promise.all([refetch(), new Promise((resolve) => setTimeout(resolve, 1000))])
```

Comment:
Since this logic is used in two separated instances, this line could be abstracted in a separated function placed in commons utils.
```typescript
/**
 * This is useful when you want to make sure that a promise takes at least a certain amount of time to resolve,
 * for example when you want to avoid flickering of a loading spinner if the promise resolves too quickly.
 */
export async function delayedPromise<TResult>(promise: Promise<TResult>, delayMs: number): Promise<TResult> {
  const [result] = await Promise.all([promise, new Promise((resolve) => setTimeout(resolve, delayMs))])
  return result
}
```

This line could become:
```typescript
delayedPromise(refetch(), 1000)
```

Reply:
If you find a better naming for the function, go for it! :)